gen/templates/resource_test.go.tmpl

```diff
@@ -4,32 +4,68 @@
 
 package provider
 
+{{- $found_cert := false -}}
+{{- range $key, $value := $.all}}
+  {{- if contains $key "cert" }}
+    {{- $found_cert = true -}}
+  {{- end}}
+{{- end}}
+
 import (
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	{{- if $found_cert }}
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	{{- end}}
 )
 
 {{- if .parents}}
 	{{- range .parents}}
 func TestAccResource{{$.resourceClassName}}With{{capitalize .class_name}}(t *testing.T) {
 
 	resource.Test(t, resource.TestCase{
-		PreCheck:                 func() { testAccPreCheck(t) },
+PreCheck: func() { 
+  time.Sleep(10 * time.Second)
+  testAccPreCheck(t, "{{.test_type}}")
+},
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
 			// Create with minimum config and verify default APIC values
 			{
 				Config: testConfig{{$.resourceClassName}}MinDependencyWith{{capitalize .class_name}},
 				ExpectNonEmptyPlan: {{.class_in_parent}},
 				Check: resource.ComposeAggregateTestCheckFunc(
+				{{- if $found_cert }}
+				func(s *terraform.State) error {
+                        stateRefreshFunc := func() (interface{}, string, error) {
+                            time.Sleep(15 * time.Second)
+                            return nil, "", nil
+                        }
+                        conf := &resource.StateChangeConf{
+                            Refresh: stateRefreshFunc,
+							Timeout: 1 * time.Second,
+                        }
+                        conf.WaitForState()
+                        return nil
```

Comment:
could you add some inline comment explaining the need of this logic?

Reply:
ok